File: apps/ruby-sdk/lib/firecrawl/models/credit_usage.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Credit usage information.
    class CreditUsage
      attr_reader :remaining_credits, :plan_credits,
                  :billing_period_start, :billing_period_end

      def initialize(data)
        @remaining_credits = data["remainingCredits"]
        @plan_credits = data["planCredits"]
        @billing_period_start = data["billingPeriodStart"]
        @billing_period_end = data["billingPeriodEnd"]
      end

      def to_s
        "CreditUsage{remaining=#{remaining_credits}/#{plan_credits}}"
      end
    end
  end
end

File: apps/.net-sdk/Firecrawl/Models/CreditUsage.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

/// <summary>
/// Account credit usage information.
/// </summary>
public class CreditUsage
{
    [JsonPropertyName("remaining_credits")]
    public int? RemainingCredits { get; set; }

    [JsonPropertyName("total_credits_used")]
    public int? TotalCreditsUsed { get; set; }

    [JsonPropertyName("billing_period_start")]
    public string? BillingPeriodStart { get; set; }

    [JsonPropertyName("billing_period_end")]
    public string? BillingPeriodEnd { get; set; }
}
